tools/clang/test/HLSLFileCheck/passes/dxil/dxil_remove_dead_pass/delete_constant_dce.ll

```diff
@@ -102,21 +108,41 @@ attributes #3 = { nounwind readonly }
-!27 = !DILocalVariable(tag: DW_TAG_auto_variable, name: "zero", scope: !4, file: !1, line: 23, type: !7)
-!28 = !DILocation(line: 24, column: 19, scope: !4)
-!29 = !DILocation(line: 24, column: 9, scope: !4)
-!30 = !DILocalVariable(tag: DW_TAG_auto_variable, name: "ret", scope: !4, file: !1, line: 24, type: !7)
-!31 = !DILocation(line: 26, column: 3, scope: !4)
+!14 = !{!"dxc(private) 1.7.0.3848 (dxil-op-cache-init, 44ba911a0)"}
+!15 = !{!"delete_constant_dce.hlsl", !"// RUN: %dxc %s -T ps_6_0 -Od | FileCheck %s\0D\0A\0D\0A// This test verifies the fix for a deficiency in RemoveDeadBlocks where:\0D\0A//\0D\0A// - Value 'ret' that can be reduced to constant by DxilValueCache is removed\0D\0A// - It held on uses for a PHI 'val', but 'val' was not removed\0D\0A// - 'val' is not used, but also not DCE'ed until after DeleteDeadRegion is run\0D\0A// - DeleteDeadRegion cannot delete 'if (foo)' because 'val' still exists.\0D\0A\0D\0A// CHECK: @main\0D\0A// CHECK-NOT: phi\0D\0A\0D\0Acbuffer cb : register(b0) {\0D\0A  float foo;\0D\0A}\0D\0A\0D\0A[RootSignature(\22\22)]\0D\0Afloat main() : SV_Target {\0D\0A  float val = 0;\0D\0A  if (foo)\0D\0A    val = 1;\0D\0A\0D\0A  float zero = 0;\0D\0A  float ret = val * zero;\0D\0A\0D\0A  return ret;\0D\0A}\0D\0A"}
+!16 = !{!"delete_constant_dce.hlsl"}
+!17 = !{!"-E", !"main", !"-T", !"ps_6_0", !"-fcgl", !"-Od", !"-Zi", !"-Qembed_debug"}
+!18 = !{i32 1, i32 0}
+!19 = !{i32 1, i32 7}
+!20 = !{!"ps", i32 6, i32 0}
+!21 = !{null, null, !22, null}
+!22 = !{!23}
+!23 = !{i32 0, %cb* @cb, !"cb", i32 0, i32 0, i32 1, i32 4, null}
+!24 = !{i32 0, %cb undef, !25}
+!25 = !{i32 4, !26}
+!26 = !{i32 6, !"foo", i32 3, i32 0, i32 7, i32 9}
+!27 = !{i32 1, void (float*)* @main, !28}
+!28 = !{!29, !30}
+!29 = !{i32 0, !2, !2}
+!30 = !{i32 1, !31, !32}
+!31 = !{i32 4, !"SV_Target", i32 7, i32 9}
+!32 = !{i32 0}
+!33 = !{void (float*)* @main, !"main", !34, !21, null}
+!34 = !{null, !35, null}
+!35 = !{!36}
+!36 = !{i32 0, !"SV_Target", i8 9, i8 16, !32, i8 0, i32 1, i8 1, i32 0, i8 0, null}
+!37 = !{[24 x i8] c"\02\00\00\00\00\00\00\00\18\00\00\00\00\00\00\00\18\00\00\00\00\00\00\00"}
+!38 = !DILocation(line: 19, column: 9, scope: !4)
+!39 = !DILocalVariable(tag: DW_TAG_auto_variable, name: "val", scope: !4, file: !1, line: 19, type: !7)
+!40 = !DIExpression()
+!41 = !DILocation(line: 20, column: 7, scope: !42)
+!42 = distinct !DILexicalBlock(scope: !4, file: !1, line: 20, column: 7)
+!43 = !DILocation(line: 20, column: 7, scope: !4)
+!44 = !DILocation(line: 21, column: 9, scope: !42)
+!45 = !DILocation(line: 21, column: 5, scope: !42)
+!46 = !DILocation(line: 23, column: 9, scope: !4)
+!47 = !DILocalVariable(tag: DW_TAG_auto_variable, name: "zero", scope: !4, file: !1, line: 23, type: !7)
+!48 = !DILocation(line: 24, column: 19, scope: !4)
+!49 = !DILocation(line: 24, column: 9, scope: !4)
+!50 = !DILocalVariable(tag: DW_TAG_auto_variable, name: "ret", scope: !4, file: !1, line: 24, type: !7)
```

Comment:
Do we need debug info for this test?

Reply:
The original test did, and it's probably good to leave in, since there are debug intrinsic uses, such as `call void @llvm.dbg.value(metadata float %val.0, i64 0, metadata !39, metadata !40), !dbg !38` that should not prevent the dead block from being eliminated.  Besides, it helps for following the IR from the original HLSL scenario.